src/controllers/category-controller.js

```diff
@@ -72,8 +74,34 @@ async function getCategory(req, res, next) {
   }
 }
 
+async function uploadCategoryImage(req, res, next) {
+  try {
+    const { file } = req;
+    const { categoryId } = req.params;
+
+    const categoryExists = await CategoryModel.exists({ _id: categoryId });
+    if (!categoryExists) {
+      throw new NotFoundError(`Category with id = ${categoryId} not found.`);
+    }
+
+    const fileType = await FileManagerLib.getFileType(file);
+
+    if (!fileType.mime) {
+      throw new ValidationError('Only images allowed');
+    }
+
+    const _id = categoryId;
+    const url = await uploadAndUpdateModelItem({ file, fileType, _id }, CategoryModel);
```

Comment:
if you separate this kind of logic in a function, make it general

for example it sets uploaded key to the 'image', but it may be the case that we re-use this function for uploading and setting to model's other field